tests/console/snapper_cleanup_timeline.pm

```diff
@@ -36,36 +36,48 @@ sub convert2numeric {
     return $str2convert;
 }
 
+# returns the percentage % of used disk space.
 sub get_used_disk_space {
     my $used_space = script_output("df -h /  | awk \'NR==2 {print \$5}\'");
     return convert2numeric($used_space);
 }
 
+# returns the disk size number without units
+sub get_disk_size {
+    my $disk_size = script_output("df -h /  | awk \'NR==2 {print \$2}\'");
+    return convert2numeric($disk_size);
+}
+
 sub run {
     my $test_data = get_test_suite_data();
     record_info("Check quota", "Verify that the percentage of root filesystem quota is less than 50%");
     my $qgroup_space = script_output("btrfs qgroup show --sync --si / | grep \"1/0\" | awk \'{print \$3}\'");
     $qgroup_space = convert2numeric($qgroup_space);
     my $space_limit = $test_data->{snapper_config}->{SPACE_LIMIT};
-    my $disk_size   = script_output("df -h /  | awk \'NR==2 {print \$2}\'");
-    $disk_size = convert2numeric($disk_size);
+    my $disk_size   = get_disk_size();
     die "Snapshots take more than " . $space_limit * 100 . "% of root disk space"
       if ($qgroup_space >= $space_limit * $disk_size);
 
+    # The cleanup timeline algorith will erase timeline snapshots if the free disk space is less than FREE_LIMIT%
     my $free_limit = $test_data->{snapper_config}->{FREE_LIMIT} * 100;
-    record_info("Space > $free_limit%", "Ensure that free space is more than $free_limit%. 
+    record_info("Free disk > $free_limit%", "Ensure that free disk space is more than $free_limit%. 
 	    Create a snapshot with timeline cleanup algorithm and make sure that it is not erased by the algorithm");
-    die "Free disk space is less than $free_limit%" if (get_used_disk_space() > 100 - $free_limit);
+    my $used_disk = get_used_disk_space();
+    die "Free disk space is less than $free_limit%" if ($used_disk > 100 - $free_limit);
     assert_script_run("snapper create --description \"timeline\" --cleanup-algorithm timeline",
         fail_message => "Timeline snapshot failed to be created");
     assert_script_run("snapper cleanup timeline",
         fail_message => "Timeline cleanup algorithm failed to run");
     assert_script_run("snapper ls | grep timeline",
         fail_message => "No timeline snapshot found");
 
-    record_info("Space < $free_limit%", "Fill up disk space and make sure that the created snapshot gets erased by 
+    record_info("Free disk < $free_limit%", "Fill up disk space and make sure that the created snapshot gets erased by 
 	    timeline cleanup algorithm");
-    assert_script_run("dd if=/dev/urandom of=/tmp/blob bs=10M count=3080", timeout => 900,
+    # Calculating the number of blocks (size 10M) that need to be filled with random data, in order to
+    # fill up the disk more than (100 - $free_limit)%. Due to rounding up of "df -h" command, the module
+    # adds 10% to (100-FREE_LIMIT)%
+    my $block_number = (100 - $free_limit - $used_disk + 10) * $disk_size;
```

Comment:
We could indicate with record_info what percentage is intended to fill up, so in this case 72% + error margin of 10% (we need so much margin?) equal to 82%. And then we could say that we are assuming disk in GB, so those percentages cancel out, giving as a result the number of blocks. Something more readable I mean, other option, even better than words, is just write the maths of one example in the comment, then it would be super clear for somebody trying to understand this part.

Reply:
From the verification runs, we need at least 7% margin. I rounded up to ten, to unpleasant surprises. So we go for (100 - 20 +10 ) = 90% full disk. I added the record_info, good point.

Reply:
what takes for me a bit to understand is that we multiply percentage by disk size and then we get number of blocks, the unit does not match at least you think a bit about the cancellation mentioned in previous comment.
In other words we could include those conversions in this formula even if they give us as result '1' and include factor '10' on those, so someone adjusting those 10MB below in `dd` would notice that this formula needs to be adjusted as well.

Reply:
I think this would just complicate it more. I will add the units in the comment, in order to explain how I ended up with this formula.